test/interpreter/function-arabic.spec.ts

```diff
@@ -0,0 +1,100 @@
+import {ErrorType, HyperFormula} from '../../src'
+import {ErrorMessage} from '../../src/error-message'
+import {adr, detailedError} from '../testUtils'
+
+describe('Function ARABIC', () => {
+  it('should return #NA! error with the wrong number of arguments', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=ARABIC()', '=ARABIC(1, 1)'],
+    ])
+
+    expect(engine.getCellValue(adr('A1'))).toEqualError(detailedError(ErrorType.NA, ErrorMessage.WrongArgNumber))
+    expect(engine.getCellValue(adr('B1'))).toEqualError(detailedError(ErrorType.NA, ErrorMessage.WrongArgNumber))
+  })
+
+  it('should properly sanitize input', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=ARABIC(" XD ")'],
+      ['=ARABIC("xd")'],
+      ['=ARABIC(" xD ")'],
+    ])
+    expect(engine.getCellValue(adr('A1'))).toEqual(490)
+    expect(engine.getCellValue(adr('A2'))).toEqual(490)
+    expect(engine.getCellValue(adr('A3'))).toEqual(490)
+  })
+
+  it('should detect incorrect numerals', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=ARABIC("IMM")'],
+      ['=ARABIC("MMMM")'],
+      ['=ARABIC("IXC")'],
+      ['=ARABIC("--I")'],
+      ['=ARABIC("-")'],
+      ['=ARABIC("Ma")'],
+      ['=ARABIC("M M")'],
+    ])
+    expect(engine.getCellValue(adr('A1'))).toEqualError(detailedError(ErrorType.VALUE, ErrorMessage.Roman))
+    expect(engine.getCellValue(adr('A2'))).toEqualError(detailedError(ErrorType.VALUE, ErrorMessage.Roman))
+    expect(engine.getCellValue(adr('A3'))).toEqualError(detailedError(ErrorType.VALUE, ErrorMessage.Roman))
+    expect(engine.getCellValue(adr('A4'))).toEqualError(detailedError(ErrorType.VALUE, ErrorMessage.Roman))
+    expect(engine.getCellValue(adr('A5'))).toEqualError(detailedError(ErrorType.VALUE, ErrorMessage.Roman))
+    expect(engine.getCellValue(adr('A6'))).toEqualError(detailedError(ErrorType.VALUE, ErrorMessage.Roman))
+    expect(engine.getCellValue(adr('A7'))).toEqualError(detailedError(ErrorType.VALUE, ErrorMessage.Roman))
+  })
+
+  it('works for border cases', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=ARABIC("MMMIMIDCCCICILXXXIXIVIII")'],
+      ['=ARABIC("-I")'],
```

Comment:
I've got another border case:
- empty string: `If an empty string ("") is used as an input value, 0 is returned.`
- case insensitive: `The case of the text argument is ignored. For example, "mxmvii" evaluates to the same result (1997) as "MXMVII."`

Source: https://support.microsoft.com/en-us/office/arabic-function-9a8da418-c17b-4ef9-a657-9370a30a674f

Reply:
but those are checked

Reply:
Ok, I see case sensitivity in sanitize section. But where is an empty string edge case covered?

Reply:
it is now :)